pkg/kubectl/cmd/logs/logs.go

```diff
@@ -307,13 +352,33 @@ func (o LogsOptions) RunLogs() error {
 	return nil
 }
 
-func DefaultConsumeRequest(request *rest.Request, out io.Writer) error {
+// DefaultConsumeRequest reads the data from request and writes into
+// the out writer. It buffers data from requests until the newline or io.EOF
+// occurs in the data, so it doesn't interleave logs sub-line
+// when running concurrently.
+//
+// A successful read returns err == nil, not err == io.EOF.
+// Because the function is defined to read from request until io.EOF, it does
+// not treat an io.EOF as an error to be reported.
+func DefaultConsumeRequest(request rest.ResponseWrapper, out io.Writer) error {
 	readCloser, err := request.Stream()
 	if err != nil {
 		return err
 	}
 	defer readCloser.Close()
 
-	_, err = io.Copy(out, readCloser)
-	return err
+	r := bufio.NewReader(readCloser)
+	for {
+		bytes, err := r.ReadBytes('\n')
+		if _, err := out.Write(bytes); err != nil {
+			return err
+		}
+
+		if err != nil {
```

Comment:
```
if err != nil && err != io.EOF {
    return err
}
return nil
```

Reply:
@soltysh `return nil` will stop the `for` loop after the first iteration. We want to return from the function when the first error appears, but we don't treat `io.EOF` as an error (because it is what we are waiting for). In case of no error we want to continue the loop.